android-huawei/src/de/golfgl/gdxgamesvcs/HuaweiGameServicesClient.java

```diff
@@ -0,0 +1,671 @@
+
+    private final int HUAWEI_GAMESVCS_AUTH_REQUEST = 8971;
+    private final int HUAWEI_GAMESVCS_ACHIEVEMENTS_REQUEST = 8972;
+    private final int HUAWEI_GAMESVCS_LEADERBOARDS_REQUEST = 8973;
+
+    private AndroidApplication activity;
+    private IGameServiceIdMapper<String> huaweiLeaderboardIdMapper;
+    private IGameServiceIdMapper<String> huaweiAchievementIdMapper;
+    private IGameServiceIdMapper<String> huaweiGameEventIdMapper;
+    private JosAppsClient josAppsClient;
+    private AchievementsClient achievementsClient;
+    private RankingsClient leaderboardsClient;
+    private EventsClient eventsClient;
+    private ArchivesClient archivesClient;
+    private IGameServiceListener gsListener;
+    private Player currentPlayer;
+
+    private boolean isSaveDataEnabled;
+    private boolean isSessionActive = false;
+
+    public HuaweiGameServicesClient(AndroidApplication activity, boolean isSaveDataEnabled) {
+        this.activity = activity;
+        this.isSaveDataEnabled = isSaveDataEnabled;
+        this.activity.addAndroidEventListener(this);
+        this.josAppsClient = JosApps.getJosAppsClient(this.activity);
+        this.josAppsClient.init();
+        this.achievementsClient = Games.getAchievementsClient(this.activity);
+        this.leaderboardsClient = Games.getRankingsClient(this.activity);
+        this.eventsClient = Games.getEventsClient(this.activity);
+        this.archivesClient = Games.getArchiveClient(this.activity);
+    }
+
+    /**
+     * sets up the mapper for leaderboard ids
+     *
+     * @param huaweiLeaderboardIdMapper
+     * @return this for method chaining
+     */
+    public HuaweiGameServicesClient setHuaweiLeaderboardIdMapper(IGameServiceIdMapper<String> huaweiLeaderboardIdMapper) {
+        this.huaweiLeaderboardIdMapper = huaweiLeaderboardIdMapper;
+        return this;
+    }
+
+    /**
+     * sets up the mapper for achievement ids
+     *
+     * @param huaweiAchievementIdMapper
+     * @return this for method chaining
+     */
+    public HuaweiGameServicesClient setHuaweiAchievementIdMapper(IGameServiceIdMapper<String> huaweiAchievementIdMapper) {
+        this.huaweiAchievementIdMapper = huaweiAchievementIdMapper;
+        return this;
+    }
+
+    /**
+     * sets up the mapper for game event ids
+     *
+     * @param huaweiGameEventIdMapper
+     * @return this for method chaining
+     */
+    public HuaweiGameServicesClient setHuaweiGameEventIdMapper(IGameServiceIdMapper<String> huaweiGameEventIdMapper) {
+        this.huaweiGameEventIdMapper = huaweiGameEventIdMapper;
+        return this;
+    }
+
+    @Override
+    public String getGameServiceId() {
+        return IGameServiceClient.GS_HUAWEI_ID;
+    }
+
+    @Override
+    public void setListener(IGameServiceListener gsListener) {
+        this.gsListener = gsListener;
+    }
+
+    @Override
+    public boolean resumeSession() {
+        return logIn();
```

Comment:
`login()` falls back to explicit log in if silent login fails. This is not what `resumeSession` should do. 
Take a look at GpgsClient ot get an idea:
https://github.com/MrStahlfelge/gdx-gamesvcs/blob/93cf6de5d8440e7a2c7fd420d75f9d299ef57453/android-gpgs/src/de/golfgl/gdxgamesvcs/GpgsClient.java#L157

https://github.com/MrStahlfelge/gdx-gamesvcs/blob/93cf6de5d8440e7a2c7fd420d75f9d299ef57453/android-gpgs/src/de/golfgl/gdxgamesvcs/GpgsClient.java#L260

Reply:
I'll resolve this conversation because Huawei doc suggests to try the silentSignIn and then to manage the error with a resolution intent, forwarding the user to the signIn/signUp page